vault/build/titusvaults/sources/Vault.move

```diff
@@ -49,64 +74,133 @@ module titusvaults::Vault {
     struct InstantWithdrawVaultEvent has drop, store {
         withdrawer: address,
         amount: u64,
-        shares_burnt: u64,
+        shares_burnt: u64
     }
 
     #[event]
     struct StandardWithdrawVaultEvent has drop, store {
         withdrawer: address,
         amount: u64,
-        shares_burnt: u64,
+        shares_burnt: u64
     }
 
-    public entry fun set_current_time(_host: &signer) {
-        let host_addr = address_of(_host);
-        assert!(host_addr == @titusvaults, E_NOT_AUTHORIZED);
-        move_to(_host, Round {
-            initial_time: timestamp::now_microseconds(),
-            round: 0,
+    #[event]
+    struct RoundUpdatedEvent has drop, store {
+        new_round_id: u64
+    }
+
+    #[event]
+    struct StrikePriceUpdatedEvent has drop, store {
+        round_id: u64,
+        strike_price: u64
+    }
+
+    #[event]
+    struct PremiumPriceUpdatedEvent has drop, store {
+        round_id: u64,
+        premium_price: u64
+    }
+
+    #[event]
+    struct MintOptionsEvent has drop, store {
+        round_id: u64,
+        amount: u64,
+        strike_price: u64,
+        premium_price: u64,
+    }
+
+    // init round state
+    public entry fun initialize_round_state(_host: &signer) {
+        move_to(_host, RoundState {
+            current_round_id: 1,
+            shares: smart_table::new(),
+            strike_price: 0,
+            premium_price: 0,
+            round_start_time: 0,
+            deposit_start_time: 0,
+            option_creation_time: 0,
+            exercise_time: 0,
+            close_timestamp: 0,
+            total_amount_deposited: 0,
+            total_premium_collected: 0,
+            timestamps_set: false,
+            is_options_minted: false
         });
     }
+    
+    // keeper functions
+    public fun update_round_from_keeper(_host: &signer) acquires RoundState, VaultMap {
+        let host_addr = signer::address_of(_host);
+        let round_state = borrow_global_mut<RoundState>(@titusvaults);
```

Comment:
what round ID state are we getting here?

Reply:
here also, perform action only on Round ID passed to the function. If a function doesn't have roundID as a parameter, accept one.